spec/acceptance/transient_attributes_spec.rb

```diff
@@ -1,15 +1,19 @@
 describe "transient attributes" do
   before do
+    define_model("Country")
     define_model("User", name: :string, email: :string)
 
     FactoryBot.define do
       sequence(:name) { |n| "John #{n}" }
 
+      factory :country
+
       factory :user do
         transient do
           four     { 2 + 2 }
           rockstar { true }
           upcased  { false }
+          country
```

Comment:
This implicitly tests that the transient association doesn't try to set `user.country=`, but I would like to see acceptance tests demonstrating that this really works.

The test cases should cover both explicit associations

```rb
transient do
  association :country
end
```

and implicit ones

```rb
transient do
  country
end
```

and then maybe

Reply:
I'll take a look at this.

Reply:
@composerinteralia: I'm constructing an acceptance test where the transient association is used to construct values in the non-transient attributes, but I'm running into some trouble with the `attributes_for` strategy. It looks like the behavior is that it will always initialize the transient attributes as nil when running this strategy, which causes the dependent attributes to fail:

```
  before do
    define_model("Country", country_code: :string)
    define_model("User", name: :string, email: :string, country_code: :string)

    FactoryBot.define do
      sequence(:name) { |n| "John #{n}" }

      factory :country do
        country_code { 'US' }
      end

      factory :user do
        transient do
          four     { 2 + 2 }
          rockstar { true }
          upcased  { false }
          country
        end

        name  { "#{FactoryBot.generate(:name)}#{' - Rockstar' if rockstar}" }
        email { "#{name.downcase}#{four}@example.com" }
        country_code { country.country_code }

        after(:create) do |user, evaluator|
          user.name.upcase! if evaluator.upcased
        end
      end
    end
  end

  context "returning attributes for a factory" do
    subject { FactoryBot.attributes_for(:user, rockstar: true) }

    it { should_not have_key(:four) }
    it { should_not have_key(:rockstar) }
    it { should_not have_key(:upcased) }
    it { should_not have_key(:country) }
    it { should     have_key(:name) }
    it { should     have_key(:email) }
    it { should     have_key(:country_code) }
  end
end
```

I can get around this issue by passing in `country: Country.new` to the `attributes_for` method, but that's ignoring the implicit association defined in the factory. Is this expected behavior, or a bug?

Reply:
```
  1) transient attributes returning attributes for a factory 
     Failure/Error: country_code { country.country_code }
     
     NoMethodError:
       undefined method `country_code' for nil:NilClass
```

Reply:
I might not have time to look at this in detail right away, but it reminds me of https://github.com/thoughtbot/factory_bot/issues/1292

Reply:
I finally got around to looking at this, and it is as I suspected related to #1292. It is pretty well baked in that `attributes_for` will not build any associations.

As I think about it, I don't know what the expected behavior is here. Would we expect when using `attributes_for` that the transient associations are also built with the `attributes_for` strategy? If so, that would also break your test case. Or would we expect the transient associations to be built all the time with the `build` strategy? That would pass the test, but would be a deviation from anything else in factory_bot.

This may extend to other strategies as well: if we are using the `create` strategy, should transient associations also be built with `create`? But then we might be writing records to the database that don't need to be there. Could we use `build` for the transient associations all the time? But what then if we are trying to use the id of the transient record?

These unanswered questions lead me to wonder whether we should proceed with this feature at all. I haven't yet seen a use case for it (https://github.com/thoughtbot/factory_bot/issues/1024 mentions that this doesn't work, but doesn't give an example for why we would want this to work). If this isn't something that solves a problem people are having, then it may not be worth the added complexity.

I don't really have any answers here. I am thinking out loud.